### PR TITLE
bump containerd to v1.5.10

### LINKED ISSUE #3711: Bump containerd version to 1.5.10
Now the containerd version <1.5.10 has security alert, so we need to bump the containerd version to 1.5.10.

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3711 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -69,6 +69,7 @@ replace (
 	github.com/Sirupsen/logrus v1.0.5 => github.com/sirupsen/logrus v1.0.5
 	github.com/Sirupsen/logrus v1.3.0 => github.com/Sirupsen/logrus v1.0.6
 	github.com/Sirupsen/logrus v1.4.0 => github.com/sirupsen/logrus v1.0.6
+	github.com/containerd/containerd => github.com/containerd/containerd v1.5.10
```
[the remark this answers is not in the document]
Comment (gy95) [reply go.mod:73]: IMHO both are OK.

[the remark this answers is not in the document]
Comment (fisherxu) [reply go.mod:73]: We need to bump the version >=v1.5.10, can just only add to require section? Avoid maintain two places.

[the remark this answers is not in the document]
Comment (fisherxu) [reply go.mod:73]: BTW in replace section, many `v0.0.0 =>` related items seem useless..?

File: go.mod [hunk 2]
```diff
@@ -9,11 +9,13 @@ require (
 	github.com/blang/semver v3.5.1+incompatible
 	github.com/cheekybits/genny v1.0.0 // indirect
 	github.com/container-storage-interface/spec v1.5.0
+	github.com/containerd/containerd v1.5.10 // indirect
 	github.com/docker/distribution v2.8.0+incompatible // indirect
 	github.com/docker/docker v20.10.2+incompatible
 	github.com/docker/go-connections v0.4.0
 	github.com/eclipse/paho.mqtt.golang v1.2.0
 	github.com/emicklei/go-restful v2.9.6+incompatible
+	github.com/garyburd/redigo v1.6.3 // indirect
```
Comment (fisherxu) [inline go.mod:18]: I tried locally, haven't added this line? Which is a archived repo.
```
--- a/go.mod
+++ b/go.mod
@@ -9,6 +9,7 @@ require (
        github.com/blang/semver v3.5.1+incompatible
        github.com/cheekybits/genny v1.0.0 // indirect
        github.com/container-storage-interface/spec v1.5.0
+       github.com/containerd/containerd v1.5.10 // indirect
        github.com/docker/distribution v2.8.0+incompatible // indirect
        github.com/docker/docker v20.10.2+incompatible
        github.com/docker/go-connections v0.4.0
@@ -29,7 +30,6 @@ require (
        github.com/mitchellh/go-ps v0.0.0-20190716172923-621e5597135b
        github.com/onsi/ginkgo v1.14.0
        github.com/onsi/gomega v1.10.3
-       github.com/opencontainers/image-spec v1.0.2 // indirect
        github.com/opencontainers/runc v1.0.3 // indirect
        github.com/paypal/gatt v0.0.0-20151011220935-4ae819d591cf
        github.com/prometheus/client_golang v1.11.0

```

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (gy95) [reply go.mod]: yeah

File: go.mod (hunk 2 above)
Comment (gy95) [reply go.mod]: can add upgrade dependency script: like https://github.com/kubernetes/kubernetes/blob/master/hack/pin-dependency.sh

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**